components/Auth/AuthProvider.tsx

```diff
@@ -10,47 +11,46 @@ import React, {
 } from 'react'
 import { auth, userRef } from 'utils/firebase'
 
-const AuthContext = createContext<{
-  user: User | null
-  loading: boolean
-  userData: UserDoc | null
-}>({
-  user: null,
-  loading: true,
-  userData: null,
-})
+type UseAuthReturn = {
+  /** User object for people who've been authenticated */
+  authUser?: User
+  isLoading: boolean
+  /** User database object for storing non-auth info such as a user's collections */
+  userData?: UserDocument
+  isModalOpen: boolean
+  onModalOpen: () => void
+  onModalClose: () => void
+}
+
+const AuthContext = createContext<UseAuthReturn>({} as UseAuthReturn)
 
-export const useUser = () => {
+export const useAuth = () => {
   return useContext(AuthContext)
 }
 
-//There is a distinction between user and userData
-//User is the data used by Firebase to distinguish different user accounts
-//userDoc is the data we use to handle list ownership
-
 export const AuthProvider: React.FC<{ children: ReactNode }> = ({
   children,
 }) => {
-  const [user, setUser] = useState<User | null>(null)
-  const [loading, setLoading] = useState(true)
-  const [userData, setUserData] = useState<UserDoc | null>(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [authUser, setAuthUser] = useState<User>()
+  const [userData, setUserData] = useState<UserDocument>()
+  const { isOpen, onOpen, onClose } = useDisclosure()
```

Comment:
Adding state for auth modal here so different components that change it are using the same state